### PR TITLE
Create def files and add first notebook

### LINKED ISSUE #49: Create a def_files directory
Provide a set of order parameters definitions files into a directory called `def_files`. 
Users will pick them to calculate the order parameters.

### PR DESCRIPTION
**Pull request by patrickfuchs**
I created the dir `def_files` and put there Berger POPC and CHARMM36UA POPC grabed from NMRlipids repo (https://github.com/NMRLipids/MATCH/tree/master/scripts/orderParm_defs). Then I added the first notebook on an analysis of a Berger POPC traj. The notebook has to be updated, especially how to download the .def file from the dir `def_files` just created.

This PR will fix #49

### CODE REVIEW

**Comment by patrickfuchs** [comment]
OK now the notebook is almost done. I only have to update the definition of STEM and to correct a few typos. Let me know if you have remarks / suggestions, and importantly if you understand the flow.

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 1]
```diff
@@ -0,0 +1,1342 @@
...
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "OPs = df[\"OP\"].iloc[18:46]\n",
+    "\n",
+    "x = np.arange(len(OPs))\n",
+    "plt.scatter(x, -OPs)\n",
```
Comment (HubLot) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:1264]: Why a minus in "-OPs"?

**Review by HubLot — COMMENTED** [review]
Great!
I understood the flow, I just have a couple questions.

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 1 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:1264]: The reason is well explained on Samuli's blog : http://nmrlipids.blogspot.com/2014/11/new-version-of-manuscript.html?showComment=1416584781727#c8832795399216522099
Most OP are negative, and so far il literature people have looked to absolute values. Thus they are used to look at OP plots that way.

**Comment by patrickfuchs** [comment]
Thanks for the review @HubLot ! I added error bars on the last plot. But by default it draws some line. I have to update that to remove the line and put regular filled circles instead.

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 1 above)
Comment (HubLot) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:1268]: oh yes indeed. I forgot about that :)

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 2]
```diff
@@ -0,0 +1,1394 @@
...
+    "!wget https://zenodo.org/record/13279/files/popc0-25ns.trr\n",
+    "!wget https://zenodo.org/record/13279/files/endCONF.gro\n",
+    "!ls"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "OK the 2 files freshly downloaded are here. Now we need one last file. It's a definition file relating generic names for each C-H bond to the atom names in the PDB file. In buildH, there is a specific directory [def_files](https://github.com/patrickfuchs/buildH/tree/first_notebook/def_files) from which you can download such a def file. (note that you can find some also on the [NMRlipids repo](https://raw.githubusercontent.com/NMRLipids/buildH/master/Berger_POPC_test_case/order_parameter_definitions_MODEL_Berger_POPC.def). Here we want to download the `.def` file corresponding to POPC Berger (beware that if we want to download it with `wget` from github, we have to add the prefix `raw.` in the URL):"
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:154]: Typo sur les parenthèses

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 3]
```diff
@@ -0,0 +1,1394 @@
...
+   ],
+   "source": [
+    "!head order_parameter_definitions_MODEL_Berger_POPC.def"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The first column is the generic C-H name, second is the residue (lipid) name, third column the carbon PDB name, last column the hydrogen PDB name. This file will tell buildH which Hs you want to rebuild, and then to calculate the order parameter on the corresponding C-Hs. It means that it is possible to provide only a subset of possible C-Hs if, for example, you only want those from the polar heads. Note that if you want to output a trajectory (not done in this notebook), the def file must contain **all** possible C-Hs.\n",
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:226]: Je ne comprends pas la phrase 'Note that if you want to output a trajectory, the def file must contain all possible C-Hs'

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 4]
```diff
@@ -0,0 +1,1394 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Launching buildH for order parameter calculation\n",
+    "\n",
+    "Now, we can launch buildH on this downloaded trajectory. For that we need a few arguments :\n",
+    "\n",
+    "- the topfile corresponding to the system passed without argument: it consists of a pdb or gro file, here `endCONF.gro`, buildH infers the topology of the system from it;\n",
+    "- the `-l` argument which tells which lipid we want to analyze ( **Note: we need to make a demo of the available lipids that buildH contains (should be in the output of the -h option** ), here `Berger_PLA` (`Berger` corresponds to the force field name used for this simulation, `PLA` is the residue name of the lipid we want to analyze in the topfile); note that if you simulated a mixture of different lipids, you will have to launch buildH separately for each lipid;\n",
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:275]: Soucis de parenthèses
et je ne comprends pas la première note

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 5]
```diff
@@ -0,0 +1,1394 @@
...
+    "- the `-d` argument corresponding to the def file we just downloaded, here `order_parameter_definitions_MODEL_Berger_POPC.def` ;\n",
+    "- the `-x` argument which is the trajectory file, here `popc0-25ns.trr`.\n",
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:277]: Seulement les .trr ?

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 6]
```diff
@@ -0,0 +1,1394 @@
...
+    "- the `-o` argument which tells buildH the output file name for soring the calculated order parameters.\n",
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:278]: Typo 'soring' => 'sToring'

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 7]
```diff
@@ -0,0 +1,1394 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Launching buildH for order parameter calculation\n",
+    "\n",
+    "Now, we can launch buildH on this downloaded trajectory. For that we need a few arguments :\n",
+    "\n",
+    "- the topfile corresponding to the system passed without argument: it consists of a pdb or gro file, here `endCONF.gro`, buildH infers the topology of the system from it;\n",
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:274]: Pas de flag pour le fichier de topologie ? pourquoi ce choix ?

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 8]
```diff
@@ -0,0 +1,1394 @@
...
+    "\n",
+    "```\n",
+    "$ buildH endCONF.gro -l Berger_PLA -d order_parameter_definitions_MODEL_Berger_POPC.def \\\n",
+    "    -x popc0-25ns.trr -o OP_POPC_Berger_25ns.out\n",
+    "```\n",
+    "\n",
+    "Or that way:\n",
+    "\n",
+    "```\n",
+    "$ buildH endCONF.gro -l Berger_PLA -d order_parameter_definitions_MODEL_Berger_POPC.def \\\n",
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:344]: Pourquoi pas rajouter un flag -v ? Plutôt que de rediriger vers /dev/null ?

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 9]
```diff
@@ -0,0 +1,1394 @@
...
+    "df[[\"CH_name\", \"OP\"]].iloc[49:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plot of the order parameter\n",
+    "\n",
+    "Now that we have loaded the data with pandas, and that we can select any part of the lipid, we can use the matplotlib module to make a nice plot. For example, we can plot the order parameter of the palmitoyl chain. Note that we select rows 18 to 45 to show only the first 15 carbons, since we do not want the final CH3 of carbon 16."
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:1266]: 'rows 18 to 45 to show only the first 15 carbons' problème de mise à jour des valeurs ?

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb [hunk 10]
```diff
@@ -0,0 +1,1394 @@
...
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.5"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 4
+}
```
Comment (ABacle) [inline docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:1394]: Un commentaire de fin pour clôturer le notebook ?

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 2 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:154]: Done

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 3 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:226]: I tried to rephrase it 😄

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 4 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:275]: This was just a reminder for me to add the available lipids when the user invike the `-h` option 😉

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 5 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:277]: Now we handle trr, xtc and dcd 😄

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 6 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:278]: Done

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 7 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:274]: Now it's implemented, thanks for the suggestion!

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 8 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:344]: We thought about that, we'll probably use some kind of *counter* (on the command line) telling the user how much of the calculation is left.

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 9 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:1266]: What I meant we the first 15 carbons of the palmitoyl chain (which has 16 carbons), thus we ignore the final CH3.

File: docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb (hunk 10 above)
Comment (patrickfuchs) [reply docs/notebooks/Berger_POPC_25ns_buildH_calcOP.ipynb:1394]: Good suggestion, thanks 👍 !

**Comment by patrickfuchs** [comment]
Almost there! I only have to update a few minor things after the new implemented features in buildH. More to come soon.

**Comment by patrickfuchs** [comment]
OK I merge this PR and update a few things in the notebook after the merge (more convenient like that).

**Pull request merged**